sql/analyzer/experimental_rules.go

```diff
@@ -86,6 +99,47 @@ func fixupAuxiliaryExprs(ctx *sql.Context, a *Analyzer, n sql.Node, scope *plan.
 				return n, transform.SameTree, err
 			}
 			ins.Source = newSource
+
+			rightSchema := make(sql.Schema, len(n.Destination.Schema())*2)
+			// schema = [oldrow][newrow]
+			for i, c := range n.Destination.Schema() {
+				rightSchema[i] = c
+				if _, ok := n.Source.(*plan.Values); !ok && len(n.Destination.Schema()) == len(n.Source.Schema()) {
+					rightSchema[len(n.Destination.Schema())+i] = n.Source.Schema()[i]
+				} else {
+					newC := c.Copy()
+					newC.Source = "__new_ins"
```

Comment:
Is this a meaningful name or just needs to not match any real table? Need a const?